tmt/base.py

```diff
@@ -2962,6 +2962,7 @@ class RunData(tmt.utils.SerializableContainer):
     steps: List[str]
     environment: EnvironmentType
     remove: bool
+    rp_uuid: Optional[str]
```

Comment:
Does it work? :)

If it does, we should make it more generic. I'd like to avoid ad hoc additions like this, `rp` can mean many things, if you decide you need to store 10 more attributes, we'd have 11 `rp_*` fields in this class, etc.

I'd go with some classes right from the beginning, but `RunData` is not ready, so let's start with dictionaries. How about something like this: `plugin_data: Dict[str, Dict[str, str]]`? The first key would be the plugin name, the nested dictionary would be owned by the plugin, store there whatever you wish. I expect something like this would materialize:

```python
plugin_data: {
  'reportportal': {
    'uuid': ...
  }
}
```

I'd introduce dataclasses later, we can do it in a standalone patch.

Reply:
Sure, it was temporary, I left it open to suggestions, perhaps on the following meeting. But I think your idea looks good, so I can redesign it that way.

Reply:
True. Just check whether the `report.data[0]` is the same report plugin, to avoid surprises. E.g. `rp_phases = [phase for phase in self.step.plan.my_run.plans[0].report.phases(ReportReportPortal)]` should give you their list - if it's empty, there was no RP report in the plan, otherwise you can pick the first item of this list and be sure it does have `launch_uuid`.

Reply:
I suppose the field can be removed from the PR, correct?

Reply:
@4N0body5 is this resolved?